### PR TITLE
Adding search bar to groups page

### PR DESCRIPTION
**Pull request by eiffelwong1**
Issue: #<Number>
https://github.com/social-plan-it/plan-it-social-web/issues/103

## Summary (1-2 sentences)

added search bar

## Details (reason and description of the changes)

used a similar method as in the search page.

### CODE REVIEW

File: app/routes/groups.tsx [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
+import { db } from '~/modules/database/db.server';
+import { type LoaderFunctionArgs } from '@remix-run/node';
+import type { Group } from '@prisma/client';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const url = new URL(request.url);
+  const query = url.searchParams.get('q');
+
+  if (!query) {
+    const groups = await db.group.findMany({ take: 24 });
```
Comment (eiffelwong1) [inline app/routes/groups.tsx:13]: inline with the useGroups() hook

File: app/routes/groups.tsx [hunk 2]
```diff
@@ -1,15 +1,67 @@
...
+  const groups = await db.group.findMany({
+    where: { name: { contains: query } },
+  });
+  return { groups };
+}
 
 export default function Group() {
-  const groups = useGroups();
+  const { groups } = useLoaderData<typeof loader>();
+  console.log('groups', groups);
```
Comment (tonydangblog) [inline app/routes/groups.tsx:25]: Looks like we forgot to remove this log :)

File: app/routes/groups.tsx [hunk 3]
```diff
@@ -1,15 +1,67 @@
...
+          <div className="flex p-2 sm:p-8 w-full mx-auto max-w-screen-2xl">
+            <input
+              type="search"
+              aria-label="Search For Events"
+              id="q"
+              name="q"
+              className="block w-full py-4 pl-10 text-xs sm:text-base text-gray-900 md:text-center lg:text-2xl border-none rounded-l-lg italic bg-gray-50 focus:ring-2 focus:outline-none focus:ring-blue-500 text-center"
+              placeholder="Search For Groups"
+              minLength={2}
+              maxLength={50}
```
Comment (tonydangblog) [inline app/routes/groups.tsx:39]: Are there any significance to these limits? Just curious, why did we chose 2 and 50?

File: app/routes/groups.tsx [hunk 4]
```diff
@@ -1,15 +1,67 @@
...
+                  strokeLinejoin="round"
+                  d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
+                />
+              </svg>
+            </button>
+          </div>
+        </Form>
         <ul className="w-full flex flex-wrap justify-center text-secondary">
-          {groups?.map((group) => {
+          {groups?.map((group: Group) => {
```
Comment (tonydangblog) [inline app/routes/groups.tsx:64]: Was TS complaining here? Prisma usually returns typed. Did it not in this case?

File: app/routes/groups.tsx [hunk 5]
```diff
@@ -1,15 +1,67 @@
-import { useGroups } from '~/hooks/useGroups';
-import { Link } from '@remix-run/react';
+import { Form, Link, useLoaderData } from '@remix-run/react';
 import { Image } from '~/components/ui/images';
 import { LinkButton } from '~/components/ui/forms';
+import { db } from '~/modules/database/db.server';
+import { type LoaderFunctionArgs } from '@remix-run/node';
+import type { Group } from '@prisma/client';
```
Comment (tonydangblog) [inline app/routes/groups.tsx:6]: NIT: Let's keep the `type` syntax consistent when we can. Maybe we can figure out how to add a lint rule for this in a future Eslint session :)

**Review by tonydangblog — APPROVED** [review]
Left some comments for minor improvements. I think everything is functionally good to go though.

File: app/routes/groups.tsx (hunk 3 above)
Comment (andrelandgraf) [reply app/routes/groups.tsx:39]: We could make those constant values that we share across our search fields maybe?

File: app/routes/groups.tsx [hunk 6]
```diff
@@ -1,15 +1,67 @@
...
 export default function Group() {
-  const groups = useGroups();
+  const { groups } = useLoaderData<typeof loader>();
+  console.log('groups', groups);
   return (
     <div className="bg-primary">
       <div className="m-auto max-w-screen-xl">
+        <Form method="GET" action="/groups" className="w-full bg-gray-800">
+          <div className="flex p-2 sm:p-8 w-full mx-auto max-w-screen-2xl">
+            <input
```
Comment (andrelandgraf) [inline app/routes/groups.tsx:31]: Could be a cool reusable component! (no blocker of course!)

**Review by andrelandgraf — APPROVED** [review]
Ship it!

**Pull request merged**